File: python/array_string/minimum_window/solution_test.py
from unittest import TestCase
from .solution import min_window


class TestMinWindow(TestCase):
    def test_repeated_character(self):
        S = "ADOBECODEBANC"
        T = "OBOC"
        self.assertEqual(min_window(S, T), "OBECO")

    def test_unique_character(self):
        S = "ADOBECODEBANC"
        T = "ABC"
        self.assertEqual(min_window(S, T), "BANC")

    def test_empty_string(self):
        S = ""
        T = "ABC"
        self.assertEqual(min_window(S, T), "")

    def test_empty_string_in_both_inputs(self):
        S = ""
        T = ""
        self.assertEqual(min_window(S, T), "")

    def test_list_inputs(self):
        S = []
        T = ["A"]
        self.assertEqual(min_window(S, T), "")

    def test_dictionary_inputs(self):
        S = {}
        T = {"A"}
        self.assertEqual(min_window(S, T), "")

File: go/array_string/min_window/solution/solution_test.go
package solution

import "testing"

func TestMinWindow(t *testing.T) {
	cases := []struct {
		name, mainString, substring, want string
	}{
		{"minimum substring with repeated character", "ADOBECODEBANC", "OBOC", "OBECO"},
		{"minimum substring with unique character", "ADOBECODEBANC", "ABC", "BANC"},
		{"use string input without match", "a", "b", ""},
		{"use empty string inputs", "", "", ""},
		{"give length of substring less than main string", "ABCD", "ADOBECODEBANC", ""},
	}

	for _, tc := range cases {
		t.Run(tc.name, func(t *testing.T) {
			got := minWindow(tc.mainString, tc.substring)

			if got != tc.want {
				t.Errorf("minWindow(%q, %q) got %q; want %q.", tc.mainString, tc.substring, got, tc.want)
			}
		})
	}
}
